sonar-scala-plugin/src/main/java/org/sonarsource/scala/checks/StringLiteralDuplicatedScalaCheck.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * SonarSource Scala
+ * Copyright (C) 2018-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the Sonar Source-Available License Version 1, as published by SonarSource SA.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the Sonar Source-Available License for more details.
+ *
+ * You should have received a copy of the Sonar Source-Available License
+ * along with this program; if not, see https://sonarsource.com/license/ssal/
+ */
+package org.sonarsource.scala.checks;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonarsource.slang.api.NativeTree;
+import org.sonarsource.slang.api.StringLiteralTree;
+import org.sonarsource.slang.api.TopLevelTree;
+import org.sonarsource.slang.api.Tree;
+import org.sonarsource.slang.checks.StringLiteralDuplicatedCheck;
+import org.sonarsource.slang.checks.api.CheckContext;
+import org.sonarsource.slang.checks.api.InitContext;
+import org.sonarsource.slang.checks.api.SecondaryLocation;
+
+@Rule(key = "S1192")
+public class StringLiteralDuplicatedScalaCheck extends StringLiteralDuplicatedCheck {
+
+
+  private static final int DEFAULT_THRESHOLD = 3;
+  private static final int MINIMAL_LITERAL_LENGTH = 5;
+  private static final Pattern NO_SEPARATOR_REGEXP = Pattern.compile("\\w++");
+  private static final String ANNOTATION_QUALIFIED_NAME = "ScalaNativeKind(class scala.meta.Mod$Annot$ModAnnotImpl)";
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(TopLevelTree.class, (ctx, tree) -> {
+      Map<String, List<StringLiteralTree>> occurrences = new HashMap<>();
+
+      var annotationsStringLiterals = tree.descendants()
+        .filter(t -> t instanceof NativeTree nativeTree && nativeTree.nativeKind().toString().equals(ANNOTATION_QUALIFIED_NAME))
+        .flatMap(Tree::descendants)
+        .filter(StringLiteralTree.class::isInstance).toList();
```

Comment:
Optional, but you could make this method lighter by extracting this stream into a separate method:

```suggestion
      var annotationsStringLiterals = getAnnotationsStringLiterals(tree);
```

```
private static List<Tree> getAnnotationsStringLiterals(TopLevelTree tree) {
    return tree.descendants()
      .filter(t -> t instanceof NativeTree nativeTree && nativeTree.nativeKind().toString().equals(ANNOTATION_QUALIFIED_NAME))
      .flatMap(Tree::descendants)
      .filter(StringLiteralTree.class::isInstance).toList();
  }
```

Reply:
Yes indeed. I extracted the method and also changed the return type to make it return a set for more efficient filtering.